dse.c:replace_inc_dec mis-use of gen_int_mode

http://gcc.gnu.org/bugzilla/show_bug.cgi?id=46337

--- gcc-4_3-branch/gcc/ChangeLog	2010/11/29 10:53:25	167237
+++ gcc-4_3-branch/gcc/ChangeLog	2010/11/29 11:03:03	167238
@@ -1,3 +1,11 @@
+2010-11-29  Eric Botcazou  <ebotcazou@adacore.com>
+
+	PR rtl-optimization/46337
+	Backport from mainline
+	2009-04-20  Ian Lance Taylor  <iant@google.com>
+
+	* dse.c (replace_inc_dec): Reverse parameters to gen_int_mode.
+
 2010-11-18  Jakub Jelinek  <jakub@redhat.com>
 
 	PR middle-end/46534
--- gcc-4_3-branch/gcc/dse.c	2010/11/29 10:53:25	167237
+++ gcc-4_3-branch/gcc/dse.c	2010/11/29 11:03:03	167238
@@ -775,7 +775,7 @@
     case POST_INC:
       {
 	rtx r1 = XEXP (x, 0);
-	rtx c = gen_int_mode (Pmode, data->size);
+	rtx c = gen_int_mode (data->size, Pmode);
 	emit_insn_before (gen_rtx_SET (Pmode, r1, 
 				       gen_rtx_PLUS (Pmode, r1, c)),
 			  data->insn);
@@ -786,7 +786,7 @@
     case POST_DEC:
       {
 	rtx r1 = XEXP (x, 0);
-	rtx c = gen_int_mode (Pmode, -data->size);
+	rtx c = gen_int_mode (-data->size, Pmode);
 	emit_insn_before (gen_rtx_SET (Pmode, r1, 
 				       gen_rtx_PLUS (Pmode, r1, c)),
 			  data->insn);
-- 
